docs/csharp/programming-guide/generics/constraints-on-type-parameters.md

```diff
@@ -10,72 +10,111 @@ helpviewer_keywords:
+
+- The `!=` and `==` operators cannot be used because there is no guarantee that the concrete type argument will support these operators.
+- They can be converted to and from `System.Object` or explicitly converted to any interface type.
+- You can compare to [null](../../language-reference/keywords/null.md). If an unbounded parameter is compared to `null`, the comparison will always return false if the type argument is a value type.
+
+## Type Parameters as Constraints
+
+The use of a generic type parameter as a constraint is useful when a member function with its own type parameter has to constrain that parameter to the type parameter of the containing type, as shown in the following example:
+
+[!code-csharp[using the class and struct constraints](../../../../samples/snippets/csharp/keywords/GenericWhereConstraints.cs#13)]
+
+In the previous example, `T` is a type constraint in the context of the `Add` method, and an unbounded type parameter in the context of the `List` class.
+
+Type parameters can also be used as constraints in generic class definitions. Note that the type parameter must be declared within the angle brackets together with any other type parameters:
+
+[!code-csharp[using the class and struct constraints](../../../../samples/snippets/csharp/keywords/GenericWhereConstraints.cs#14)]
+
+The usefulness of type parameters as constraints with generic classes is very limited because the compiler can assume nothing about the type parameter except that it derives from `System.Object`. Use type parameters as constraints on generic classes in scenarios in which you want to enforce an inheritance relationship between two type parameters.
+
+## Unmanaged constraint
+
+Beginning with C# 7.3, you can use the `unmanaged` constraint to specify that the type parameter must be an **unmanaged types**. An **unmanaged type** is a type which is not a reference type and doesn't contain reference type fields at any level of nesting. The `unmanaged` constraint enables you to write reusable routines to work with types that can be manipulated as blocks of memory, as shown in the following example:
+
+[!code-csharp[using the unmanaged constraint](../../../../samples/snippets/csharp/keywords/GenericWhereConstraints.cs#15)]
+
+The preceding method must be compiled in an `unsafe` context because it uses the `sizeof` operator on a type not known to be a builtin type. Without the `unmanaged` constraint, the `sizeof` operator would be unavailable.
+
+## Delegate constraints
+
+Also beginning with C# 7.3, you can use `System.Delegate` or `System.MulticastDelegate` as a base class constraint. The CLR always allowed this, but the C# language disallowed it. The `System.Delegate` constraint enables you to write code that works with delegates in a type safe manner. The following code defines an extension method that combines two delegates provided they are the same type:
```

Comment:
The second sentence is an interesting fact, but I'm not sure it adds information that makes using constraints easier. Should we remove it, and, if not, then to add the same remark for enum constraints?

Reply:
I added it in both places. It clarifies that a new CLR is not needed.